### PR TITLE
wip: 0.2

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/langchain/langchain/chains/loading.py
Comment (baskaryan) [inline libs/langchain/langchain/chains/loading.py:1]: break

File: libs/langchain/langchain/memory/chat_memory.py
Comment (baskaryan) [inline libs/langchain/langchain/memory/chat_memory.py:1]: break

File: libs/langchain/langchain/memory/kg.py [hunk 1]
```diff
@@ -19,18 +15,49 @@
...
+    @classmethod
+    def from_string(cls, triple_string: str) -> "KnowledgeTriple":
+        """Create a KnowledgeTriple from a string."""
+        subject, predicate, object_ = triple_string.strip().split(", ")
+        subject = subject[1:]
+        object_ = object_[:-1]
+        return cls(subject, predicate, object_)
+
+
+def _kg_default_factory() -> Any:
```
Comment (baskaryan) [inline libs/langchain/langchain/memory/kg.py:37]: break

File: libs/langchain/langchain/memory/zep_memory.py
Comment (baskaryan) [inline libs/langchain/langchain/memory/zep_memory.py:1]: break

File: libs/langchain/langchain/retrievers/web_research.py
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/web_research.py:1]: break

File: libs/community/langchain_community/indexes/graph.py
Comment (baskaryan) [inline libs/community/langchain_community/indexes/graph.py:1]: move that wasn't already marked as deprecated

Comment (baskaryan) [inline libs/community/langchain_community/indexes/graph.py:1]: moved GraphIndexCreator to community

File: libs/core/langchain_core/tools.py
Comment (baskaryan) [inline libs/core/langchain_core/tools.py:1]: moved BaseToolkit to core

File: libs/langchain/langchain/chains/api/openapi/chain.py
Comment (baskaryan) [inline libs/langchain/langchain/chains/api/openapi/chain.py:1]: need to figure out what to do with dependency on Requests

File: libs/langchain/langchain/chains/llm_requests.py
Comment (baskaryan) [inline libs/langchain/langchain/chains/llm_requests.py:1]: break

File: libs/langchain/langchain/chains/loading.py
Comment (baskaryan) [inline libs/langchain/langchain/chains/loading.py:1]: break

File: libs/langchain/langchain/chains/openai_functions/openapi.py
Comment (baskaryan) [inline libs/langchain/langchain/chains/openai_functions/openapi.py:1]: break

File: libs/langchain/langchain/memory/entity.py [hunk 2]
```diff
@@ -153,6 +158,8 @@ class RedisEntityStore(BaseEntityStore):
 
     Entities get a TTL of 1 day by default, and
     that TTL is extended by 3 days every time the entity is read back.
+
+    Must have `redis` and `langchain-community` installed.
```
Comment (hwchase17) [inline libs/langchain/langchain/memory/entity.py:162]: this should be moved to community, no?

File: libs/langchain/langchain/memory/zep_memory.py [hunk 3]
```diff
@@ -13,22 +11,32 @@ class ZepMemory(ConversationBufferMemory):
...
-                    url=ZEP_API_URL,        # Your Zep server's URL
-                    api_key=<your_api_key>, # Optional
-                    memory_key="history",   # Ensure this matches the key used in
-                                            # chain's prompt template
-                    return_messages=True,   # Does your prompt template expect a string
-                                            # or a list of Messages?
-                )
-        chain = LLMChain(memory=memory,...) # Configure your chain to use the ZepMemory
-                                              instance
+            !pip install -U langchain langchain-community
```
Comment (hwchase17) [inline libs/langchain/langchain/memory/zep_memory.py:21]: this should be moved to community?

File: libs/langchain/langchain/memory/chat_memory.py
Comment (baskaryan) [inline libs/langchain/langchain/memory/chat_memory.py:1]: break

File: libs/langchain/langchain/memory/entity.py (hunk 2 above)
Comment (baskaryan) [reply libs/langchain/langchain/memory/entity.py:162]: yea but BaseEntityStore isn't in community yet. could move, just not something we've deprecated yet

File: libs/langchain/langchain/retrievers/self_query/base.py
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/self_query/base.py:1]: ugly

File: libs/langchain/langchain/retrievers/self_query/redis.py [hunk 4]
```diff
@@ -2,17 +2,6 @@
 
 from typing import Any, Tuple
 
-from langchain_community.vectorstores.redis import Redis
```
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/self_query/redis.py:5]: need to figure out what to do with all these

File: libs/langchain/langchain/tools/vectorstore/tool.py
Comment (baskaryan) [inline libs/langchain/langchain/tools/vectorstore/tool.py:1]: probably should have never moved this to community, for now think we should have duplicate versions

File: libs/langchain/langchain/memory/zep_memory.py (hunk 3 above)
Comment (baskaryan) [reply libs/langchain/langchain/memory/zep_memory.py:21]: BaseMemory is in langchain

**Pull request closed without merging**